data/modules/Debug/DebugCommodityPrices.lua

```diff
@@ -175,24 +160,27 @@ local scan_systems = function(dist)
 			-- table.sort(comm, function (a,b) return a:GetName() > b:GetName() end)
 
 			-- include goods if legal, or if we've chosen to include it
-			local include = include_illegal or sys:IsCommodityLegal(comm.name)
+			local include = sys:IsCommodityLegal(comm.name)
+			if include_illegal then
+				include = true
+			end
 
-			-- TODO: also include negative products, right?
-			-- if comm.price > 0 then
+			if include_only_purchasable then
+				include = include and comm.purchasable
+			end
 
-				local name = comm:GetName()
-				local price = getprice(comm, sys)
+			local name = comm:GetName()
+			local price = getprice(comm, sys)
 
-				if not commodities[name] then
-					commodities[name] = Commodity:new(name, comm.price)
-					print(name, comm.price)
-				end
+			if include and not commodities[name] then
+				commodities[name] = Commodity:new(name, comm.name, comm.price)
+				print("Adding:", name, comm.name, comm.price)
```

Comment:
Should this print statement remain? If the debug UI is working I don't think it's necessary to validate the results.

Reply:
I'll remove it. Had some debugging that was driving me crazy.